src/app/mods/health-bar/health-bar.ts

```diff
@@ -1,153 +1,148 @@
-            .lifeBarsContainer {
-                position: absolute;
-                top: 0;
-                left: 0;
-                pointer-events: none;
-                z-index: 1;
-                visibility: hidden;
-            }
 
-            .lifeBarContainer {
-                box-sizing: border-box;
-                border: 1px gray solid;
-                background-color: #222;
-                height: 6px;
-                width: 80px;
-                position: absolute;
-                border-radius: 3px;
-                overflow: hidden;
-                transition-duration: 500ms;
-                margin-top: 10px;
-            }
+    startMod(): void {}
 
-            .lifeBar {
-                transition-duration: 300ms;
-                height: 100%;
-                width: 0%;
-                background-color: #333;
-            }
+    private bars: { [fighterId: number]: Bar; } = { };
+    private container: HTMLDivElement;
 
-            .shieldBar {
-                transition-duration: 300ms;
-                height: 100%;
-                width: 0%;
-                margin-left: 50%;
-                background-color: #944ae0;
-                position: absolute;
-                top: 0;
-            }
+    constructor(
+        wGame: any,
+        settings: SettingsService,
+        translate: TranslateService
+    ){
+        super(wGame,settings,translate);
 
-            .lifePointsText {
-                font-size: 12px;
-                position: absolute;
-                width: 80px;
-                color: white;
-                text-shadow: 0 0 5px rgba(0, 0, 0, 0.9);
-                transition-duration: 500ms;
-                margin-top: 16px;
-                margin-left: 2px;
-            }`;
-            this.wGame.document.getElementsByTagName('head')[0].appendChild(healthbarCss);
+        if (this.settings.option.vip.general.health_bar) this.enableHealthBars();
+    }
 
-            this.shortcutsHelper = new ShortcutsHelper(this.wGame);
-            this.barContainer = new BarContainer(this.wGame);
+    private enableHealthBars(){
+        Logger.info('- enable Health-Bar');
 
+        let healthbarCss = document.createElement('style');
+        healthbarCss.id = 'healthbarCss';
+        healthbarCss.innerHTML = `
 
-            this.removeOnDeath();
-            this.setFightStart();
-            this.displayOnStart();
-            this.stopOnFightEnd();
-            this.stopOnFightStop();
+        .lifeBarsContainer {
+            position: absolute;
+            top: 0;
+            left: 0;
+            pointer-events: none;
+            z-index: 1;
+        }
 
+        .lifeBarContainer {
+            box-sizing: border-box;
+            background-color: #222;
+            height: 6px;
+            width: 80px;
+            position: absolute;
+            border-radius: 3px;
+            overflow: hidden;
+            transition-duration: 500ms;
```

Comment:
Duration but no transition-property ?

Reply:
Object changes position so the transition-duration is used(i changed the value for 300 tho)